Gum/Commands/EditCommands.cs

```diff
@@ -465,6 +466,81 @@ public void DuplicateSelectedElement()
+            var instances = SelectedState.Self.SelectedInstances.ToList();
+            if (instances == null || instances.Count == 0 || element == null)
+            {
+                MessageBox.Show("You must first save the project before adding a new component");
+            }
+            else if (instances is List<InstanceSave>)
+            {
+                TextInputWindow tiwcw = new TextInputWindow();
+                tiwcw.Message = "Enter new Component name:";
+
+                FilePath filePath = element.Name;
+                var nameWithoutPath = filePath.FileNameNoPath;
+
+                tiwcw.Result = nameWithoutPath;
+                //tiwcw.Option = $"Replace {nameWithoutPath} and all children with an instance of the new component";
+
+                if (tiwcw.ShowDialog() == DialogResult.OK)
+                {
+                    string name = tiwcw.Result;
+                    //bool replace = tiwcw.Checked
+
+                    string whyNotValid;
+                    NameVerifier.Self.IsComponentNameValid(tiwcw.Result, "", null, out whyNotValid);
+
+                    if (string.IsNullOrEmpty(whyNotValid))
+                    {
+                        ComponentSave componentSave = new ComponentSave();
+                        componentSave.BaseType = "Container";
+                        string folder = null;
+                        if (!string.IsNullOrEmpty(folder))
+                        {
+                            folder += "/";
+                        }
+                        componentSave.Name = folder + name;
+
+                        StateSave defaultState;
+
+                        // Clone instances
+                        foreach (var instance in instances)
+                        {
+                            var instanceSave = instance.Clone();
+                            instanceSave.BaseType = instance.BaseType;
+                            instanceSave.ParentContainer = componentSave;
+
+                            componentSave.Instances.Add(instanceSave);
+                        }
+
+                        // Clone states
+                        foreach (var state in element.States)
+                        {
+                            if (element.DefaultState == state)
+                            {
+                                defaultState = state.Clone();
+
+                                componentSave.Initialize(defaultState);
+                                continue;
+                            }
+                            componentSave.States.Add(state.Clone());
+                        }
+
+                        StandardElementsManagerGumTool.Self.FixCustomTypeConverters(componentSave);
+                        ProjectCommands.Self.AddComponent(componentSave);
+                        
+                    }
+                    else
+                    {
+                        MessageBox.Show($"Invalid name for new component: {whyNotValid}");
```

Comment:
Should we re-open the dialog to let the user correct the problem? Otherwise if they have an error, they have to do the right-click command again.